### PR TITLE
IPAddress: Use BinaryPrimitives.ReverseEndianness

### PR DESCRIPTION
**Pull request by justinvp**
Went ahead and grabbed this follow-up from #26303.

### CODE REVIEW

File: src/System.Net.Primitives/src/System/Net/IPAddress.cs [hunk 1]
```diff
@@ -401,11 +402,7 @@ public static long HostToNetworkOrder(long host)
 #if BIGENDIAN
```
Comment (jkotas) [inline src/System.Net.Primitives/src/System/Net/IPAddress.cs]: Nit: You can also replace this ifdef with `BitConverter.IsLittleEndian` check to make this code portable while you are on it.

`BitConverter.IsLittleEndian` is JIT intrinsic, so the native code should be identical.

Comment (justinvp) [reply src/System.Net.Primitives/src/System/Net/IPAddress.cs]: Fixed

**Comment by jkotas** [comment]
Thanks!

**Comment by stephentoub** [comment]
If/how did this affect perf?

**Comment by justinvp** [comment]
> If/how did this affect perf?

```
      Method |      Mean |     Error |    StdDev |
------------ |----------:|----------:|----------:|
 Int16Before | 2.2111 ns | 0.0191 ns | 0.0159 ns |
  Int16After | 0.9696 ns | 0.0127 ns | 0.0119 ns |
 Int32Before | 2.1893 ns | 0.0270 ns | 0.0253 ns |
  Int32After | 0.9376 ns | 0.0132 ns | 0.0124 ns |
 Int64Before | 2.8212 ns | 0.0290 ns | 0.0257 ns |
  Int64After | 0.9023 ns | 0.0100 ns | 0.0089 ns |
```

**Comment by stephentoub** [comment]
Great, thanks.

**Pull request merged**